tutorial/source/svi_part_iii.ipynb

```diff
@@ -282,52 +323,51 @@
     "        # conditionally independent given f and get vectorization\n",
     "        with pyro.iarange(\"data_iarange\"):\n",
     "            # observe all ten datapoints using the bernoulli likelihood\n",
-    "            pyro.sample(\"obs\", dist.Bernoulli(...), obs=self.data)\n",
+    "            pyro.sample(\"obs\", dist.Bernoulli(f), obs=self.data)\n",
     "\n",
     "    def guide(self, use_decaying_avg_baseline):\n",
     "        # register the two variational parameters with pyro\n",
-    "        log_alpha_q = pyro.param(\"log_alpha_q\", self.log_alpha_q_0)\n",
-    "        log_beta_q = pyro.param(\"log_beta_q\", self.log_beta_q_0)\n",
-    "        alpha_q, beta_q = torch.exp(log_alpha_q), torch.exp(log_beta_q)\n",
+    "        alpha_q = pyro.param(\"alpha_q\", torch.tensor(self.alpha_q_0),\n",
+    "                             constraint=constraints.positive)\n",
+    "        beta_q = pyro.param(\"beta_q\", torch.tensor(self.beta_q_0),\n",
+    "                            constraint=constraints.positive)\n",
     "        # sample f from the beta variational distribution\n",
     "        baseline_dict = {'use_decaying_avg_baseline': use_decaying_avg_baseline,\n",
     "                         'baseline_beta': 0.90}\n",
     "        # note that the baseline_dict specifies whether we're using\n",
     "        # decaying average baselines or not\n",
-    "        pyro.sample(\"latent_fairness\", dist.Beta(alpha_q, beta_q),\n",
+    "        f=pyro.sample(\"latent_fairness\", NonreparameterizedBeta(alpha_q, beta_q),\n",
```

Comment:
Did you add `f=` for debugging purposes? It does not seem to be used anywhere.

Reply:
good catch, fixed

Reply:
push?